planner/core/rule_build_key_info.go

```diff
@@ -96,7 +102,8 @@ func (p *LogicalSelection) BuildKeyInfo(selfSchema *expression.Schema, childSche
 	p.baseLogicalPlan.BuildKeyInfo(selfSchema, childSchema)
 	for _, cond := range p.Conditions {
 		if sf, ok := cond.(*expression.ScalarFunction); ok && sf.FuncName.L == ast.EQ {
-			if p.checkMaxOneRowCond(sf.GetArgs()[0], sf.GetArgs()[1], childSchema[0]) || p.checkMaxOneRowCond(sf.GetArgs()[1], sf.GetArgs()[0], childSchema[0]) {
+			isOk := isNullRejected(p.ctx, selfSchema, sf)
```

Comment:
Oh I didn't realize that the `ast.EQ` was checked in line 104 already, so we don't need to check `isNullRejected` anymore, since it is always true. Sorry for the previous misleading.

Reply:
done